stapi_fastapi/models/product.py

```diff
@@ -21,20 +22,41 @@ class Provider(BaseModel):
     url: AnyHttpUrl
 
 
-class Product(BaseModel):
-    type: Literal["Product"] = "Product"
+class ProductMeta(BaseModel):
```

Comment:
As an implementer I find the mental overhead of having to understand why I need to use `ProductMeta` instead of the more obvious `Product` rather confusing. Also, I would still want the option to be able to specify links for a `Product`, such as for external product documentation, license text, or other such related static content. As a result, I feel like the split of `Product` and `ProductMeta` here is artificially limiting, and I would propose you revert back to a single `Product` class. 

Of course the motivation here remains: how to properly handle the links you need to inject into the response models?

I think you can solve this by retaining the same pattern implemented here, except instead of transforming `ProductMeta` into a `Product`, you instead transform one `Product` instance into a new `Product` instance and add the links to the latter. Maybe like this:

```
class Product(BaseModel):
    type: Literal["Product"] = "Product"
    conformsTo: list[str] = Field(default_factory=list)
    id: str
    title: str = ""
    description: str = ""
    keywords: list[str] = Field(default_factory=list)
    license: str
    providers: list[Provider] = Field(default_factory=list)
    links: list[Link]
    parameters: JsonSchemaModel

    def with_links(self: Self, links: list[Link] | None = None) -> Self:
        if not links:
            return self

        attribs = copy(meta.__dict__)
        attribs.links.extend(links)
        return type(self)(**attribs)
```

Then where you are currently calling `Product.from_meta(product, links)` you'd instead be able to call `product.with_links(links)`.

What do you think of this approach?

---

Caveat: I'm not in love this solution, but it appears a pragmatic approach. It seems like the copy shouldn't be necessary, and instead it would be best support something like callable link functions that resolve during model serialization (as serialization is in effect itself a copy). Such a scheme would also allow implementers to inject dynamic links into the initial product link set that could be resolved based on the `request` context, if required.

That said, actually implementing a serialization-based approach seems like it might be hard at best or convoluted at worst (hurdles immediately apparent include how to define the pydantic typing, making sure the jsonschema model generation makes sense, and uncertainty around how best to manage the serialization process). So, given premature optimization, evil, yada yada, whatever, I say keep it easier for now, hence the above suggestion.